### PR TITLE
resolve node v14 failing tests

### PR DESCRIPTION
**Pull request by iambumblehead**
resolve errors in node v14 pipelines

### CODE REVIEW

**Comment by iambumblehead** [comment]
@aladdin-add would mind reviewing approving and/or giving your thoughts?

I did try to avoid putting something like this into the script, but I don't believe there is a way to detect which call needs to be used -- the 'older' calls causes a runtime error in new versions of node when loaders are chained together, the 'newer' call breaks older versions of node.

**Comment by iambumblehead** [comment]
@aladdin-add the problem area had a lengthy descriptive comment above it here https://github.com/iambumblehead/esmock/blob/master/src/esmockLoader.js#L36-L50

**Comment by iambumblehead** [comment]
closes https://github.com/iambumblehead/esmock/issues/87

**Comment by iambumblehead** [comment]
@aladdin-add let's merge and publish this now to resolve node14 issues. If you have any suggestions or concerns I can follow up with those tomorrow (its late at night here)

File: src/esmockLoader.js [hunk 1]
```diff
@@ -15,6 +15,9 @@ const urlDummy = 'file:///' + path
   .join(path.dirname(url.fileURLToPath(import.meta.url)), 'esmock.js')
   .replace(/^\//, '');
 
+const isLT1612 = ' 16. 12' > process.versions.node.split('.')
+  .slice(0, 2).map(s => s.padStart(3)).join('.');
```
Comment (aladdin-add) [inline src/esmockLoader.js:19]: I just like comparing numbers than strings, if you don't want the package node-semver. 😄 
```js
const [major, minor, patch] = process.versions.node.split('.').map(it => +it);

const isLT1612 = major < 16 || (major === 16 && minor < 12);
```

Comment (iambumblehead) [reply src/esmockLoader.js:19]: @aladdin-add I like this suggestion

Comment (iambumblehead) [reply src/esmockLoader.js:19]: notes for myself, for tomorrow
```diff
 -// later versions of node v16 include 'node-addons'
 +// later versions of node v14 and v16 include 'node-addons'
```

**Pull request merged**